pkg/gits/git_cli.go

```diff
@@ -1223,6 +1223,11 @@ func (g *GitCLI) CherryPickTheirs(dir string, commitish string) error {
 	return g.gitCmd(dir, "cherry-pick", commitish, "--strategy=recursive", "-X", "theirs")
 }
 
+// CherryPickTheirsKeepRedundantCommits does a git cherry-pick of commit
+func (g *GitCLI) CherryPickTheirsKeepRedundantCommits(dir string, commitish string) error {
```

Comment:
Would it make sense to have a test case? It might also help to understand what we try to fix here.

Reply:
or maybe it would be enough to understand:

> jx upgrade boot fails with the following error if there are duplicate commit's

how come there are duplicate commits? How does one get into this state?

Reply:
That is a good question. From what I can see....
A PR is created containing a commit. Another PR is then created containing a commit with the same change as the first. Both PR's are merged.
When cherry-picking the commits the second one throws an error as in the description. (btw. adding --allow-empty to the cherry-pick command didn't work.)

Reply:
Thanks for the clarification.